torchci/lib/cacheGithubAPI.ts

```diff
@@ -0,0 +1,77 @@
+// Various utilities for caching GitHub API responses
+
+import { createClient } from "@clickhouse/client";
+import dayjs from "dayjs";
+import * as thisModule from "./cacheGithubAPI";
+import { queryClickhouse } from "./clickhouse";
+
+export function enableCache() {
+  // Used for testing
+  return true;
+}
+
+function getClickhouseClient() {
+  return createClient({
+    host: process.env.CLICKHOUSE_HUD_USER_URL ?? "http://localhost:8123",
+    username: process.env.CLICKHOUSE_HUD_USER_WRITE_USERNAME ?? "default",
+    password: process.env.CLICKHOUSE_HUD_USER_WRITE_PASSWORD ?? "",
+  });
+}
+
+async function saveCache(key: string, data: any) {
```

Comment:
how would you know if the cached data is out of date? PR labels is something that can easily go out of date

Also, given CH's append only design, would cached items remain in the cache until the table's retention period expires?

Reply:
It gets invalidated in invalidateCacheBot when a pull_request gets updated, hopefully this includes labels, I'll go check